src/__tests__/geojson-layer.test.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import GeoJSONLayer from '../geojson-layer';
+import { withContext } from 'recompose';
+import { mount } from 'enzyme';
+
+describe('GeoJSONLayer', () => {
+  let GeoJSONLayerWithContext: any;
+  let addLayerMock = jest.fn();
+
+  beforeEach(() => {
+    addLayerMock = jest.fn();
+
+    GeoJSONLayerWithContext = withContext({
+      map: React.PropTypes.object
+    }, () => ({
+      map: {
+        addSource: jest.fn(),
+        addLayer: addLayerMock
+      }
+    }))(GeoJSONLayer);
+  });
+
+  it('Should call addLayer with provided layerOptions', () => {
+    mount(
+      <GeoJSONLayerWithContext
+        fillPaint={{'fill-color': '#001122'}}
+        data={{ "type": "FeatureCollection", "features": []}}
+        layerOptions={{minzoom: 13}}
+      /> as React.ReactElement<any>
+    );
+
+    const addFillLayerCall = addLayerMock.mock.calls.find(([ call ]) => call.id.endsWith('-fill'));
+    expect(addFillLayerCall).toEqual([{
+      id: 'geojson-1-fill',
+      source: 'geojson-1',
+      type: 'fill',
+      layout: {visibility: 'visible'},
+      minzoom: 13,
+      paint: {'fill-color': '#001122'}
+    }, undefined]);
+  });
+
+  it('Should call addLayer when no layerOptions provided', () => {
+    mount(
+      <GeoJSONLayerWithContext
+        fillPaint={{'fill-color': '#001122'}}
+        data={{ "type": "FeatureCollection", "features": []}}
```

Comment:
Can you define a const for `fillPaint` and `data` and use it for both tests ? Also the linter doesn't check the test I will fix it but can you use single quote 👍